synapse/federation/sender/per_destination_queue.py

```diff
@@ -92,6 +92,23 @@ def __init__(
         self._destination = destination
         self.transmission_loop_running = False
 
+        # True whilst we are sending events that the remote homeserver missed
+        # because it was unreachable.
+        # False whilst we are known to not be catching up.
+        # None when it has not been determined if we are catching-up yet (we
+        # do so lazily).
+        # New events will only be sent once this is finished, at which point
+        # _catching_up is flipped to False.
+        self._catching_up = None  # type: Optional[bool]
```

Comment:
rather than having `_catching_up` be a tristate (which feels fragile to me, cf the fact that we drop the first PDU afaict), could we start off with this set to `False`, and use a different way of saying that we need to check whether to do a catchup (for example: `_last_successful_stream_ordering is None`) ?